src/pretix/base/services/orders.py

```diff
@@ -2008,6 +2008,20 @@ def set_addons(self, addons, limit_main_positions=None):
                     for a in current_addons[cp][k][:current_num - input_num]:
                         if a.canceled:
                             continue
+                        is_unavailable = (
+                            # If an item is no longer available due to time, it should usually also be no longer
+                            # user-removable, because e.g. the stock has already been ordered.
+                            # We always pass has_voucher=True because if a product now requires a voucher, it usually does
+                            # not mean it should be unremovable for others.
```

Comment:
> But does it mean it should be auto-removed when the order is changed?

No, probably not ;)

> or display them in a way where they can be optionally removed, but their amount can't be increased?

Yes, that would be best, but it's … quite nasty to implement. We've done it for the quota case already, so it's possible.

> Not a blocker as this doesn't change the current behaviour, but isn't this the same bug just triggered by a different setting?

…yes. It will be very hard to find all the possible edge cases here :)

Reply:
I'll still merge this for now since it's still an improvement